pkg/utils/packet.go

```diff
@@ -0,0 +1,176 @@
+package utils
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"net"
+	"syscall"
+
+	"golang.org/x/net/icmp"
+	"golang.org/x/net/ipv6"
+)
+
+var (
+	arpPacketName    = "ARP"
+	icmpV6PacketName = "ICMPv6"
+)
+
+// htons converts an uint16 from host to network byte order.
+func htons(i uint16) uint16 {
+	return (i<<8)&0xff00 | i>>8
+}
+
+// formatPacketFieldWriteError builds an error string for the cases when writing to a field of a packet fails.
+func formatPacketFieldWriteError(field string, packetType string, writeErr error) error {
+	return fmt.Errorf("failed to write the %s field in the %s packet: %v", field, packetType, writeErr)
+}
+
+// IsValidMACAddress checks if net.HardwareAddr is a valid MAC address.
+func IsValidMACAddress(addr net.HardwareAddr) bool {
```

Comment:
WDYT about moving IsValidMacAddress, IsIPv4 and IsIPv6 to utils.go ?

keeping here just the bits which deal solely with sending GARP/NA packets

Reply:
I think that's a good idea. Made the change.

Reply:
Thanks for reviewing my changes. Much appreciated!